src/tr1/game/ui/elements/bar.c

```diff
@@ -0,0 +1,135 @@
+#include "game/ui/elements/bar.h"
+
+#include "game/output.h"
+
+#include <libtrx/config.h>
+#include <libtrx/game/scaler.h>
+#include <libtrx/game/ui/helpers.h>
+#include <libtrx/utils.h>
+
+#include <math.h>
+#include <string.h>
+
+#define M_BORDER 2
+#define M_PADDING 2
+#define M_COLOR_STEPS 5
+
+typedef struct {
+    UI_BAR_SETTINGS settings;
+} M_DATA;
+
+#define C(c)                                                                   \
```

Comment:
Just checking this naming is as intended (I'm fine with the super shorthand and it gets undeffed, plus it's obvious what it does, it's just a bit of a deviation I feel).

Reply:
Thanks for calling it out – I found a more elegant way to do this I think.